numpy/core/src/multiarray/methods.c

```diff
@@ -1381,6 +1381,7 @@ array_argsort(PyArrayObject *self, PyObject *args, PyObject *kwds)
             return NULL;
         }
         newd = PyArray_DescrNew(saved);
+        Py_DECREF(newd->names);
```

Comment:
Shouldn't this be `Py_XDECREF` or can `name` never be `NULL`?

Reply:
This is the same as in `sort`. I do not think the name can be possibly be `NULL` for structured dtypes. And it will only go into this branch if it is a structured dtype.